src/modules/Notifies/X11Notify.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+	QMPlay2 is a video and audio player.
+	Copyright (C) 2010-2016  Błażej Szczygieł
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Lesser General Public License as published
+	by the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Lesser General Public License for more details.
+
+	You should have received a copy of the GNU Lesser General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <X11Notify.hpp>
+#include <QMPlay2Core.hpp>
+#include <Module.hpp>
+
+#include "dbus/notification.h"
+
+#include <QImage>
+
+#include <QDBusArgument>
+
+#define X11NotifyName "X11 Notify"
+
+QDBusArgument &operator<<(QDBusArgument &arg, const QImage &image)
+{
+	if (image.isNull())
+	{
+		// Sometimes this gets called with a null QImage for no obvious reason.
+		arg.beginStructure();
+		arg << 0 << 0 << 0 << false << 0 << 0 << QByteArray();
+		arg.endStructure();
+		return arg;
+	}
+	const QImage scaled = image.scaledToHeight(100, Qt::SmoothTransformation).convertToFormat(QImage::Format_ARGB32);
+#if Q_BYTE_ORDER == Q_LITTLE_ENDIAN
+	// ABGR -> ARGB
+	const QImage i = scaled.rgbSwapped();
+#else
+	// ABGR -> GBAR
```

Comment:
Is it tested, does it make any sense (QMPlay2 doesn't support big endian - never tested)?